### PR TITLE
beacon, util, gui: Fix small error in beacon db for renewals and fix snapshot download functionality

### PR DESCRIPTION
**Pull request by jamescowens**
This PR addresses two items that are going to go into a very small leisure release in the next few days to address a duplicated renewal record in the beacon db leveldb storage on a wallet restart after the leveldb state is reapplied and the contract replay occurs.

This error could result a node getting stuck if a restart were to occur on a block that contains a renewal contract for a beacon, and upon restart, the wallet reorganizes to a block prior to the block that contains that contract, in which case the renewal record would be removed when the renewal contract is encountered again.

The solution is to fix the offending code that was creating the duplicate record error and also version the beacon database leveldb storage. Running a wallet with this code will detect an out of date beacon leveldb storage and automatically rebuild it from a contract replay to remove any bad records created by this problem.

Before this fix, if the wallet got stuck on a reorg after restart, the solution would be to restart it again, and it would proceed.

To see if a wallet has been affected by duplicate records, one can run auditsnapshotaccruals. If the duplicate record problem exists, auditsnapshotaccruals will crash the wallet. The solution will be to install the upgrade with this fix, or start the wallet with the -clearbeaconhistory option to force a rebuild of the beacon records from the contract replay.

It also implements functionality to remove the lock on the registry.dat file of the accrual repository to allow the blockchain file cleaup to proceed properly after the snapshot download.

### CODE REVIEW

**Comment by jamescowens** [comment]
This attachment documents the investigation of the beacon db problem and the testing of the fix.
[beacon problem fix test.log](https://github.com/gridcoin-community/Gridcoin-Research/files/6096329/beacon.problem.fix.test.log)
[beacon problem test and recovery.log](https://github.com/gridcoin-community/Gridcoin-Research/files/6096330/beacon.problem.test.and.recovery.log)

File: src/gridcoin/beacon.cpp [hunk 1]
```diff
@@ -989,6 +997,35 @@ int BeaconRegistry::BeaconDB::Initialize(PendingBeaconMap& m_pending, BeaconMap&
...
+        CTxDB txdb("r");
+
+        std::pair<std::string, std::string> key = std::make_pair("beacon_db", "version");
+
+        bool status = txdb.ReadGenericSerializable(key, version);
+
+        if (!status) version = 0;
+    }
+
+    if (version != m_version)
```
Comment (cyrossignol) [inline src/gridcoin/beacon.cpp]: I don't see why we need the `BeaconDB::m_version` member field. It looks like we want to check... 

```c++
if (version != CURRENT_VERSION) { ... }
```

...and set the version in `clear_leveldb()` with...

```c++
status &= txdb.WriteGenericSerializable(key, CURRENT_VERSION);
```

Is there a reason for the mutable state? Maybe I'm missing something...

Comment (jamescowens) [reply src/gridcoin/beacon.cpp]: I originally had it that way and was getting linker errors.

Comment (jamescowens) [reply src/gridcoin/beacon.cpp]: I will try and change it back.

Comment (jamescowens) [reply src/gridcoin/beacon.cpp:1013]: Your suggestion to change static constexpr to static const worked. !@$$% linker...

Comment (jamescowens) [reply src/gridcoin/beacon.cpp]: Fixed.

**Review by cyrossignol — APPROVED** [review]

**Pull request merged**